### DISCUSSION TITLE
--tesseract-timeout=0 and skipping ocr

### DISCUSSION THREAD

**Discussion by nikitar** [ocrmypdf/OCRmyPDF#1266] [General]
I'm looking at the [Don’t actually OCR my PDF](https://ocrmypdf.readthedocs.io/en/latest/cookbook.html#don-t-actually-ocr-my-pdf) section of the docs and trying that, but I'm still getting ocr'd results. (I'm checking by opening pdf in chrome, no text selection before and text selection after. I can also tell it's doing ocr because it takes a while on multipage documents)

Do I misunderstand the api?

The command: `ocrmypdf --tesseract_timeout=0  --output-type=pdf  in.pdf out.pdf`
Version: 16.0.3
Platform: macos 14.3.1

**Comment by nikitar** [comment]
Additionally, the docs _appear_ to conflict with regards to skipping ocr. Here is a section called [Optimize images without performing OCR](https://ocrmypdf.readthedocs.io/en/latest/cookbook.html#optimize-images-without-performing-ocr), which recommends `--skip-text` in addition to `--tesseract-timeout=0`. (Because otherwise the invocation will fail on a document with text)

In the [Advanced](https://ocrmypdf.readthedocs.io/en/latest/advanced.html#when-ocr-is-skipped) section, however, it says no image processing takes place when `--skip-text` is used. That seems to conflict with the "optimize images" part above.
> If --skip-text is issued, then no image processing or OCR will be performed on pages that already have text. The page will be copied to the output.

Is one of those incorrect, or is there extra context I am missing?

**Reply by jbarlow83** [reply]
Image preprocessing is not optimization, at least not in the terminology I've adopted.

Image preprocessing is deskewing and cleaning. I used to calling that image preprocessing everywhere through, but it looks a bit odd if you have OCR turned off to say that the file was preprocessed then it's done, so the label shown is "image processing".

Optimization is reducing image size after performing OCR (because we don't want to degrade the images in any way before OCR). 

`--skip-text --tesseract-timeout=0` ensures no image processing (just optimization)

I suppose I should probably add new CLI drivers sometime to optimizemypdf (?) processmypdf (too generic?) or something. But I cringe at the obvious name. To 2024 eyes, even ocrmypdf reads as a bit of a silly name, so I'm left with the question of what to call these things. 

The optimization is also not a general PDF optimizer; it's an optimizer that's designed around optimizing scanned images after you've done OCR with some fairly specific assumptions that aggressive JPEG and lossy PNG compression are better than downsampling (they are, usually - lossy compression can be thought of as a variable resolution downsampling, that ideally concentrates on salient features at the expense of less relevant ones). There's a lot of possible optimization opportunities not taken.

**Reply by nikitar** [reply]
Thank you, it makes sense! As for the name, I think if the tool is useful, people don't mind the name one way or another. And this seems like a very useful too, though I'm still new to it.

Regarding the original question, is that a bug, or do I misunderstand the api? I also tried version 14.0.4, and that one does seem to skip ocr when using the same two flags.

**Reply by nikitar** [reply]
[Resolved](https://github.com/ocrmypdf/OCRmyPDF/discussions/1270#discussioncomment-8687477), a local misconfiguration
